### PR TITLE
infra/gcp: mv clusters to terraform

### PR DESCRIPTION
**Pull request by spiffxp**
This changes a few things within infra/gcp
```
clusters                           -> terraform
terraform/projects/{project}       -> terraform/{project}
terraform/{project}/{cluster}/*.tf -> terraform/{project}/*.tf
```

The intent is:
- the name "terraform" better fits what this folder is... terraform modules used to manage (some of) our infrastructure
- less nesting means less directory traversal to get to where I need to `terraform apply`
- make the cluster folder just about cluster resources (possible readmes/scripts for cluster)
- make the project folder the site for all .tf files for the project

I can remove the script I used to generate this PR, but thought I'd check it in, assuming I'd have to do a fair amount of rebasing

### CODE REVIEW

**Comment by spiffxp** [comment]
/hold
To avoid forcing a bunch of rebases.  I'm happy to keep rebasing this.

I at least want https://github.com/kubernetes/k8s.io/pull/2235 to merge first

**Comment by ameukam** [comment]
/lgtm
/hold
Remove hold when ready.

File: infra/gcp/terraform/k8s-infra-prow-build-trusted/prow-build-trusted/README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ gcloud container clusters get-credentials \
 git clone git://github.com/kubernetes/k8s.io
 
 # deploy the resources; note boskos-resources.yaml isn't a configmap
-cd k8s.io/infra/gcp/clusters/k8s-infra-prow-build-trusted/prow-build-trusted
+cd k8s.io/infra/gcp/terraform/k8s-infra-prow-build-trusted/prow-build-trusted
```
Comment (MushuEE) [inline infra/gcp/terraform/k8s-infra-prow-build-trusted/prow-build-trusted/README.md:33]: In some places it looks like the uri collapsed ` infra/gcp/terraform/k8s-infra-prow-build-trustedprow-build-trusted/resources` but here it wasn't

**Comment by spiffxp** [comment]
I opted to break this before updating the jobs to listen to different paths.  So now that I've seen them fail...
- update prow jobs to listen to and use new paths: https://github.com/kubernetes/test-infra/pull/23163

**Comment by ameukam** [comment]
Ref: https://github.com/kubernetes/k8s.io/issues/516

**Pull request merged**